esphome/components/adc/adc_sensor_esp32.cpp

```diff
@@ -78,12 +78,14 @@ void ADCSensor::dump_config() {
     }
   }
   ESP_LOGCONFIG(TAG, "  Samples: %i", this->sample_count_);
+  ESP_LOGCONFIG(TAG, "  Sampling mode: %hhu", static_cast<uint8_t>(this->sampling_mode_));
```

Comment:
```suggestion
  const char* samplingModeStr;
  switch(this->sampling_mode_) {
    case SamplingMode::AVG:
      samplingModeStr = "average";
      break;
    case SamplingMode::MIN:
      samplingModeStr = "minimum";
      break;
    case SamplingMode::MAX:
      samplingModeStr = "maximum";
      break;
    default:
      samplingModeStr = "unknown";
  }
  ESP_LOGCONFIG(TAG, "  Sampling mode: %s", samplingModeStr);
```

Something like this maybe? Just a number in log looks weird imo

Reply:
Makes sense, thanks.